### PR TITLE
Binary classifier for Mental_Rotation articles

### PR DESCRIPTION
**Pull request by natsheh**

### CODE REVIEW

File: classifier.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	for i in related_ndx:
+	    results.append(inverse_ndx[str(count_pos + i)])
+	return np.unique(results)
+
+if __name__ == "__main__" :
+	parser = argparse.ArgumentParser()
+	# the file contains the documents decomposed vectors (model) to load
+	parser.add_argument("--vectors", default="results/output_svd.pickle", type=str)
+	parser.add_argument("--inv_ndx", default="results/output_paragraph_inverse_index.json", type=str)
+	parser.add_argument("--count_pos", default=183, type=int)
```
Comment (lmartinet) [inline classifier.py]: better use cnt_positive instead of count_pos or add comments to explain what it is.

File: classifier.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	results = []
+	for i in related_ndx:
+	    results.append(inverse_ndx[str(count_pos + i)])
+	return np.unique(results)
+
+if __name__ == "__main__" :
+	parser = argparse.ArgumentParser()
+	# the file contains the documents decomposed vectors (model) to load
+	parser.add_argument("--vectors", default="results/output_svd.pickle", type=str)
+	parser.add_argument("--inv_ndx", default="results/output_paragraph_inverse_index.json", type=str)
```
Comment (lmartinet) [inline classifier.py]: Please, choose a better name or comment to explain what it is.

File: classifier.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	print 'The model (document decomposed vectors) was loaded successfully'
+	vec_size = svd_vectors.shape[1]
+	pos_lst = svd_vectors[:count_pos]
+	neg_lst = svd_vectors[count_pos:]
+
+	scores = np.zeros(n_iterations)
+	for i in range(n_iterations):
+		if verbose > 0:
+			print 'iteration', i + 1 , 'out of ', n_iterations
+		train, test = train_test_split(pos_lst)
```
Comment (lmartinet) [inline classifier.py]: Add the percentage of test set, out of the annotated data. Could be passed as a parameter.

File: classifier.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+		raise NameError('Unrecognized class type. \
+		       Must be either "positive" or "negative"')
+	labels = labels.reshape(-1, 1)
+	sample = np.hstack((sample_lst, labels))
+	return sample
+
+def balanced_dataset(pos_lst, neg_lst, size):
+	pos_size = size/2
+	neg_size = size - pos_size # in case of odd int value of size
+	pisitives = sampling(pos_lst, 'positive', pos_size)
```
Comment (lmartinet) [inline classifier.py]: Misspelling variable: positive

File: classifier.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+	negatives = sampling(neg_lst, 'negative', neg_size)
+	return np.vstack((pisitives,negatives))
```
Comment (lmartinet) [inline classifier.py]: Misspelling variable: positive

File: classifier.py [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+
+def search_top_relevant(sim_results, inverse_ndx, sim_th):
```
Comment (lmartinet) [inline classifier.py]: You should add count_pos as an argument too.

**Review by lmartinet — COMMENTED** [review]
Please, take into account my comments.

**Comment by natsheh** [comment]
@lmartinet I have reflected your comments. Please check if all fine.
